controller/controller_client.go

```diff
@@ -189,11 +197,20 @@ func (c *clientHandler) handleUnitAdded(unit *client.Unit) {
 
 // handle the UnitChangedModified event from the V2 API
 func (c *clientHandler) handleUnitUpdated(unit *client.Unit) {
-	state, logLvl, _ := unit.Expected()
-	c.log.Debugf("Got unit updated for ID %s (%s/%s)", unit.ID(), state.String(), logLvl.String())
+	state, logLvl, cfg := unit.Expected()
+	c.log.Infof("Got unit updated for ID %s (%s/%s)", unit.ID(), state.String(), logLvl.String())
+	currentUnit := c.units.GetInputUnit(unit.ID(), unit.Type())
+	// check to see if only the log level needs updating
+	if onlyLogLevelUpdated(cfg.Source.AsMap(), currentUnit.config, logLvl) {
+		c.log.Infof("unit %s got update with only log level changing. Updating to %s", unit.ID(), config.ZapFromUnitLogLevel(logLvl))
+		logp.SetLevel(config.ZapFromUnitLogLevel(logLvl))
```

Comment:
The input and output unit log levels can be set independently, does this account for that? If the output unit were changed to debug but the input units left at info what would happen? Is it deterministic?

Reply:
I'm honestly not sure what the logic should be. Considering that the input units are just copies of what's sent to the inputs, should we just ignore them and only update the log level based on the output unit?

Reply:
In an ideal world they would use two separate loggers. Outside of that, you want to take the lowest level of any unit as the global log level. This way if you enable debug logs for anything you are guaranteed to actually get debug logs for it (in addition to debug logs for everything else in the process).

Reply:
Alright, tinkered with it, so now we just listen to log updates from the output unit, which makes a tad more sense, since a log update from an input unit might not really be "for" us.

Reply:
Yeah that makes sense, the agent is generating the input units itself so likely those log levels will just reflect the global agent log level all the time since nobody can edit them manually.